src/irmin-pack/irmin_pack.ml

```diff
@@ -548,6 +555,53 @@ struct
           Commit.CA.clear_lru (snd (commit_t t))
         in
         Contents.CA.sync ~clear_lrus (contents_t t)
+
+      let migrate v1 v2 =
+        Log.debug (fun l -> l "migrate");
+        let contents = contents_t v1 in
+        let nodes = node_t v1 |> snd in
+        let commits = commit_t v1 |> snd in
+        let f (k, (offset, length, m)) =
+          match m with
+          | 'B' ->
+              contents_t v2
+              |> Contents.CA.migrate_to_current_version ~offset ~length k
+                   contents
+          | 'N' | 'I' ->
+              node_t v2
+              |> snd
+              |> Node.CA.migrate_to_current_version ~offset ~length k nodes
+          | 'C' ->
+              commit_t v2
+              |> snd
+              |> Commit.CA.migrate_to_current_version ~offset ~length k commits
+          | _ -> invalid_arg "unknown content type"
```

Comment:
```suggestion
          | c -> Fmt.failwith "Unknown content type: %c" c
```

I think we should either have a `Corrupted_store` exception or just `failwith` here.